source/common/http/conn_manager_utility.cc

```diff
@@ -279,9 +279,9 @@ void ConnectionManagerUtility::mutateXfccRequestHeader(HeaderMap& request_header
   // the XFCC header.
   if (config.forwardClientCert() == ForwardClientCertType::AppendForward ||
       config.forwardClientCert() == ForwardClientCertType::SanitizeSet) {
-    const std::string uri_san_local_cert = connection.ssl()->uriSanLocalCertificate();
-    if (!uri_san_local_cert.empty()) {
-      client_cert_details.push_back("By=" + uri_san_local_cert);
+    const auto uri_sans_local_cert = connection.ssl()->uriSanLocalCertificate();
+    if (!uri_sans_local_cert.empty()) {
+      client_cert_details.push_back("By=" + uri_sans_local_cert[0]);
```

Comment:
Is there a reason that we distinguish the first SAN entry @PiotrSikora?

Reply:
I suspect it's because it's commonly only one URI SAN (eg SPIFFE does this), but there's nothing preventing someone from putting multiple URI SANs into the cert

Reply:
Fair enogh.